gr-network/lib/tcp_sink_impl.cc

```diff
@@ -159,7 +157,7 @@ void tcp_sink_impl::connect(bool initial_connection)
             d_acceptor = new asio::ip::tcp::acceptor(
                 d_io_context, asio::ip::tcp::endpoint(asio::ip::tcp::v4(), d_port));
     } else {
-        d_io_context.reset();
+        d_io_context.restart();
```

Comment:
I had to look this up, but it's exactly according to spec:
![image](https://github.com/user-attachments/assets/c81dab0b-1042-4ecb-8489-ba54ffb5e755)

https://think-async.com/Asio/boost_asio_1_30_2/doc/html/boost_asio/reference/io_context.html